bindings/java/c/host.h

```diff
@@ -0,0 +1,48 @@
+#include <jni.h>
+#include "evmc/evmc.h"
+
+#ifndef _Included_org_ethereum_evmc_Host
+#define _Included_org_ethereum_evmc_Host
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+JavaVM *jvm;
+struct evmc_host_context{
+    int index;
+};
+
+bool account_exists_fn(struct evmc_host_context*, const evmc_address*);
+
+evmc_bytes32 get_storage_fn(struct evmc_host_context*, const evmc_address*, const evmc_bytes32*);
+
+enum evmc_storage_status set_storage_fn(struct evmc_host_context*, const evmc_address*, const evmc_bytes32*, const evmc_bytes32*);
+
+evmc_uint256be get_balance_fn(struct evmc_host_context*, const evmc_address*);
+
+size_t get_code_size_fn(struct evmc_host_context*,const evmc_address*);
+
+evmc_bytes32 get_code_hash_fn(struct evmc_host_context*, const evmc_address*);
+
+size_t copy_code_fn(struct evmc_host_context*, const evmc_address*, size_t, uint8_t*, size_t);
+
+void selfdestruct_fn(struct evmc_host_context*, const evmc_address*, const evmc_address*);
+
+struct evmc_result call_fn(struct evmc_host_context*, const struct evmc_message*);
+
+struct evmc_tx_context get_tx_context_fn(struct evmc_host_context*);
+
+evmc_bytes32 get_block_hash_fn(struct evmc_host_context*, int64_t);
+
+void emit_log_fn(struct evmc_host_context*, const evmc_address*, const uint8_t*, size_t, const evmc_bytes32[], size_t);
+
+JNIEnv* attach();
+
+void detach(JNIEnv*);
+
+const struct evmc_host_interface* get_host_interface();
```

Comment:
This is the only function which needs to be public, everything else is local to `host.c`.